### PR TITLE
orcid: util to get putcodes for author works

### PR DESCRIPTION
**Pull request by szymonlopaciuk**
Signed-off-by: Szymon Łopaciuk <szymon.lopaciuk@cern.ch>

## Checklist:

- [ ] I have all the information that I need (if not, move to `RFC` and look for it).
- [ ] I linked the related issue(s) in the corresponding commit logs.
- [ ] I wrote [good commit log messages](https://github.com/torvalds/subsurface-for-dirk/blob/5f15ad5a86ada3c5e574041a5f9d85235322dabb/README#L92-L119).
- [ ] My code follows the code style of this project.
- [ ] I've added any new docs if API/utils methods were added.
- [ ] I have updated the existing documentation accordingly.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

**Comment by bavaria95** [comment]
ci test please

File: inspirehep/modules/orcid/utils.py [hunk 1]
```diff
@@ -0,0 +1,139 @@
...
+"""ORCID utils."""
+
+from __future__ import absolute_import, division, print_function
+
+import re
+
+from orcid import MemberAPI
+
+from inspire_utils.record import get_value
+from inspire_utils.config import load_config
```
Comment (iulianav) [inline inspirehep/modules/orcid/utils.py]: These should be in alphabetical order

File: tests/unit/orcid/fixtures/works.json [hunk 2]
```diff
@@ -0,0 +1,141 @@
...
+            ],
+            "last-modified-date": {
+                "value": 1516716538439
+            }
+        }
+    ],
+    "last-modified-date": {
+        "value": 1516722056845
+    }
+}
```
Comment (iulianav) [inline tests/unit/orcid/fixtures/works.json]: I would suggest a more explicit name for the fixture.

File: tests/unit/orcid/test_orcid_utils.py [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+import pytest
+import requests_mock
+
+from mock import patch
+
+from inspirehep.modules.orcid.utils import get_author_putcodes, CONFIG
+
+
+def get_file(fixture_name):
+    """Get and parse an XML file from ./fixtures directory"""
```
Comment (iulianav) [inline tests/unit/orcid/test_orcid_utils.py]: Is it an XML file or a JSON file?

File: tests/unit/orcid/fixtures/work.json [hunk 4]
```diff
@@ -0,0 +1,151 @@
...
+                "journal-title": {
+                    "value": "4th RESCEU International Symposium on Birth and Evolution of the Universe"
+                },
+                "path": null,
+                "short-description": null,
+                "type": "CONFERENCE_PAPER"
+            }
+        }
+    ]
+}
```
Comment (iulianav) [inline tests/unit/orcid/fixtures/work.json]: I would say it is always good to have a new line at the end of a file. It is also consistent with the rest of the the project.

File: tests/unit/orcid/fixtures/works.json (hunk 2 above)
Comment (szymonlopaciuk) [reply tests/unit/orcid/fixtures/works.json]: Done :+1:

File: tests/unit/orcid/test_orcid_utils.py [hunk 5]
```diff
@@ -0,0 +1,79 @@
...
+from __future__ import absolute_import, division, print_function
+
+import os
+import pkg_resources
+import pytest
+import requests_mock
+
+from mock import patch
+
+from inspirehep.modules.orcid.utils import get_author_putcodes, CONFIG
```
Comment (drjova) [inline tests/unit/orcid/test_orcid_utils.py]: why not ``from inspire_utils.config import load_config```? :)

File: inspirehep/modules/orcid/utils.py [hunk 6]
```diff
@@ -0,0 +1,139 @@
...
+from inspire_utils.record import get_value
+
+
+RECID_FROM_INSPIRE_URL = re.compile(
+    r"https?://(?:labs\.)?inspirehep\.net/(?:record|literature)/(\d+)",
+    re.IGNORECASE
+)
+
+WORKS_BULK_QUERY_LIMIT = 50
```
Comment (drjova) [inline inspirehep/modules/orcid/utils.py]: IMO this will be more helpful if it's part of the configuration, WDYT?

File: inspirehep/modules/orcid/utils.py [hunk 7]
```diff
@@ -0,0 +1,139 @@
...
+
+    Returns:
+        MemberAPI: ORCID API
+    """
+    client_key = CONFIG['ORCID_APP_CREDENTIALS']['consumer_key']
+    client_secret = CONFIG['ORCID_APP_CREDENTIALS']['consumer_secret']
+    return MemberAPI(client_key, client_secret)
+
+
+def _split_lists(sequence, chunk_size):
```
Comment (drjova) [inline inspirehep/modules/orcid/utils.py]: shall we add some unit tests specific for this function? There are many corner cases

File: inspirehep/modules/orcid/utils.py (hunk 6 above)
Comment (szymonlopaciuk) [reply inspirehep/modules/orcid/utils.py]: Idk, this constant will not change, unless there's a new version of ORCID API in which case we'd probably need to change the logic here too. Is it worth polluting the config file with such a thing? I made a constant for it, to not have a cryptic 50 hanging in the code...

Comment (drjova) [reply inspirehep/modules/orcid/utils.py]: Since it's a constant we can leave it as it is! 👍

**Comment by szymonlopaciuk** [comment]
ci test please

File: tests/unit/orcid/test_orcid_utils.py [hunk 8]
```diff
@@ -0,0 +1,98 @@
...
+        ([1, 2, 3, 4, 5], 3, [[1, 2, 3], [4, 5]]),
+        (['a', 'b', 'c'], 1, [['a'], ['b'], ['c']]),
+        (['just_one'], 1, [['just_one']]),
+        (['just_one'], 10, [['just_one']]),
+        ([], 10, []),
+    ]
+)
+def test_split_lists(test_sequence, test_length, expected):
+    result = _split_lists(test_sequence, test_length)
+    assert expected == result
```
Comment (drjova) [inline tests/unit/orcid/test_orcid_utils.py]: new line here :)

**Comment by david-caro** [comment]
ci merge please

**Comment by bavaria95** [comment]
ci merge please

**Comment by bavaria95** [comment]
ci merge please

**Comment by bavaria95** [comment]
ci merge please

**Comment by bavaria95** [comment]
ci merge please

**Comment by bavaria95** [comment]
ci merge please

**Comment by david-caro** [comment]
ci merge please

**Comment by david-caro** [comment]
Dammit xd `ERROR: Permission to szymonlopaciuk/inspire-next.git denied to jenkins-inspire.`

**Comment by david-caro** [comment]
@szymonlopaciuk can you enable the ' Allow edits from maintainers.' checkbox?

**Comment by szymonlopaciuk** [comment]
@david-caro It was already set...

**Comment by david-caro** [comment]
ci merge please

**Comment by david-caro** [comment]
ci merge please

**Comment by david-caro** [comment]
ci merge please

**Comment by david-caro** [comment]
@bavaria95 🥇 ! 🎉
